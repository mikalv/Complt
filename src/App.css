.App-bar {
  position: fixed !important;
  top: 0;
  right: 0;
}
.App-content {
  margin-top: 64px;
}
