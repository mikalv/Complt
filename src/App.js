/* eslint-disable  no-undef */
import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { List } from 'material-ui/List';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { pink500, pink700, tealA200 } from 'material-ui/styles/colors';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import NavigationMenu from 'material-ui/svg-icons/navigation/menu';
import IconButton from 'material-ui/IconButton';
import AddItem from './components/AddItem';
import Item from './components/Item';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: pink500,
    primary2Color: pink700,
    accent1Color: tealA200,
  },
});

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
      drawerOpen: false,
      drawerDocked: false,
    };
    this.logAddItem = this.logAddItem.bind(this);
    this.toggleDrawer = this.toggleDrawer.bind(this);
    this.handleItemAvatarTap = this.handleItemAvatarTap.bind(this);
    window.setTimeout(() => {
      this.windowResize();
    }, 0);
    window.addEventListener('resize', () => {
      this.windowResize();
    });
  }
  logAddItem(item) {
    const items = this.state.items;
    items.push(item);
    this.setState({ items });
  }
  toggleDrawer() {
    this.setState({ drawerOpen: !this.state.drawerOpen });
  }
  windowResize() {
    if (window.innerWidth >= 768) {
      this.setState({ drawerOpen: true, drawerDocked: true });
    } else {
      this.setState({ drawerOpen: false, drawerDocked: false });
    }
  }
  handleItemAvatarTap(e, id) {
    const items = this.state.items;
    if (items[id].isProject) {
      return;
    }
    items[id].isCompleted = !items[id].isCompleted;
    this.setState({ items });
  }
  render() {
    return (
      <MuiThemeProvider muiTheme={muiTheme}>
        <div>
          <AppBar
            title="Oak" className="drawer-margin"
            onLeftIconButtonTouchTap={this.toggleDrawer}
            iconElementLeft={this.state.drawerDocked ?
              <div /> : <IconButton disableTouchRipple><NavigationMenu /></IconButton>}
          />
          <Drawer
            docked={this.state.drawerDocked}
            onRequestChange={open => this.setState({ drawerOpen: open })}
            open={this.state.drawerOpen}
            swipeAreaWidth={50}
            zDepth={this.state.drawerDocked ? 0 : 2}
          >Drawer</Drawer>
          <div className="container drawer-margin">
            <AddItem onAdd={this.logAddItem} />
            <br />
            <List>
              {this.state.items.map((item, i) =>
                <Item id={i} key={i} item={item} onAvatarTouchTap={this.handleItemAvatarTap} />)}
            </List>
          </div>
        </div>
      </MuiThemeProvider>
    );
  }
}

export default App;
