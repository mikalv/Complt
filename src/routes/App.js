import React from 'react';
import Menu from 'react-icons/lib/md/menu';
import Sync from 'react-icons/lib/md/sync';
import Inbox from 'react-icons/lib/md/inbox';
import Assignment from 'react-icons/lib/md/assignment';
import Person from 'react-icons/lib/md/person';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';
import Snackbar from 'react-md/lib/Snackbars';
import Link from 'react-router/lib/Link';
import Button from 'react-md/lib/Buttons';
import { connect } from 'react-redux';
import mapDispatchToProps from '../utils/mapDispatchToProps';
import UpdateItemDialog from '../components/UpdateItemDialog';

const navItems = [{
  to: '/inbox',
  key: 'inbox',
  primaryText: 'Inbox',
  component: Link,
  leftIcon: <Inbox size={24} />,
}, {
  to: '/projects',
  key: 'projects',
  primaryText: 'Projects',
  component: Link,
  leftIcon: <Assignment size={24} />,
}, {
  to: '/account',
  key: 'account',
  primaryText: 'Account',
  component: Link,
  leftIcon: <Person size={24} />,
}];

export function navItemsWithActive(items, route) {
  return items.map((item) => {
    if (route === item.to) {
      return {
        ...item,
        active: true,
      };
    }
    return item;
  });
}

export const App = props => (
  <NavigationDrawer
    header={<div />}
    temporaryIconChildren={<Menu />}
    navItems={navItemsWithActive(navItems, props.location.pathname)}
    mobileDrawerType={NavigationDrawer.DrawerTypes.TEMPORARY}
    tabletDrawerType={NavigationDrawer.DrawerTypes.FULL_HEIGHT}
    desktopDrawerType={NavigationDrawer.DrawerTypes.FULL_HEIGHT}
    toolbarTitle="Complt"
    toolbarActions={[
      <Button icon onClick={props.sync}><Sync /></Button>,
    ]}
  >
    {props.children}
    <Snackbar toasts={props.toasts} onDismiss={props.dismissToast} />
    {props.dialogs.updateItem.visible ? <UpdateItemDialog /> : null}
  </NavigationDrawer>
);

App.propTypes = {
  children: React.PropTypes.node,
  location: React.PropTypes.shape({
    pathname: React.PropTypes.string.isRequired,
  }),
  dismissToast: React.PropTypes.func,
  sync: React.PropTypes.func,
  /* eslint-disable react/no-unused-prop-types */
  toasts: React.PropTypes.arrayOf(React.PropTypes.shape({
    text: React.PropTypes.string.isRequired,
    action: React.PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.shape({
        onClick: React.PropTypes.func,
        label: React.PropTypes.string.isRequired,
      }),
    ]),
  })).isRequired,
  dialogs: React.PropTypes.shape({
    updateItem: React.PropTypes.shape({
      visible: React.PropTypes.bool,
    }),
  }),
  /* eslint-enable react/no-unused-prop-types */
};

function mapStateToProps({ toasts, dialogs }) {
  return { toasts, dialogs };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
