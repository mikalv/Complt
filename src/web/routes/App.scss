.syncing-spinner-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 5px;
}
.App-toolbar-section {
  overflow: visible;
}
#syncing-spinner {
  margin: 0;
}
#items-to-show-select {
  width: 120px !important;
}
.App {
  height: 100%;
}
main {
  display: flex;
}
@media (min-width: 600px) {
  .drawer-margin {
    margin-left: 240px;
  }
}
.mdc-permanent-drawer__content {
  overflow: auto;
  height: 100%;
}
.App-content {
  height: 100%;
}
