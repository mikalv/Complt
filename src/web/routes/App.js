import { h, Component } from 'preact';
import { getCurrentUrl, route } from 'preact-router';
import Menu from 'react-icons/lib/md/menu';
import Sync from 'react-icons/lib/md/sync';
import Toolbar from 'preact-material-components/Toolbar';
import Drawer from 'preact-material-components/Drawer';
import { connect } from 'preact-redux';
import Async from '../components/Async';
import Spinner from '../components/Spinner';
import IconButton from '../components/IconButton';
import AppRouter from '../AppRouter';
import NavItems from '../components/NavItems';
import MatchMedia from '../MatchMedia';
import { values as itemsToShowValues } from '../../common/redux/itemsToShow';
import mapDispatchToProps from '../../common/utils/mapDispatchToProps';
import './App.scss';

const Dialogs = Async(() =>
  import(/* webpackChunkName: "dialogs" */ '../components/Dialogs')
);

const redirects = [
  ['project/inbox', '/inbox'],
  ['project/root', '/projects'],
  ['', '/inbox'],
];

const importMaterialSelectPromise = import(
  /* webpackChunkName: "mdc-select" */ '@material/select'
);

export class App extends Component {
  componentDidMount() {
    const currentUrl = getCurrentUrl();
    this.performRedirects(currentUrl);
    importMaterialSelectPromise.then(({ MDCSelect }) => {
      this.selectMDComponent = new MDCSelect(this.select);
      this.selectMDComponent.listen('MDCSelect:change', () => {
        const newValue = this.selectMDComponent.value;
        if (newValue !== this.props.itemsToShow) {
          this.props.changeItemsToShow(newValue);
        }
      });
      this.updateSelectedIndex();
    });
  }
  componentDidUpdate(prevProps) {
    this.updateSelectedIndex(prevProps);
  }
  onRouteChange = ({ url }) => {
    window.ga('set', 'page', url);
    window.ga('send', 'pageview');
    this.performRedirects(url);
  };
  onMenuButtonClick = () => {
    this.drawer.MDComponent.open = !this.drawer.MDComponent.open;
  };
  performRedirects = url => {
    redirects.forEach(redirect => {
      if (redirect[0] === url.replace(/(^\/+|\/+$)/g, ''))
        route(redirect[1], true);
    });
  };
  updateSelectedIndex = prevProps => {
    if (
      this.selectMDComponent !== undefined &&
      (prevProps && prevProps.itemsToShow) !== this.props.itemsToShow
    ) {
      const newIndex = itemsToShowValues.indexOf(this.props.itemsToShow);
      if (newIndex !== this.selectMDComponent.selectedIndex) {
        this.selectMDComponent.selectedIndex = newIndex;
      }
    }
  };
  render(props) {
    return (
      <div className="App">
        <Toolbar fixed>
          <Toolbar.Row>
            <Toolbar.Section align-start className="App-toolbar-section">
              {props.matches
                ? null
                : <IconButton
                    title={`${this.drawer && this.drawer.MDComponent.open ? 'Close' : 'Open'} navigation drawer`}
                    onClick={this.onMenuButtonClick}
                  >
                    <Menu />
                  </IconButton>}
              <div
                className="mdc-select"
                role="listbox"
                id="items-to-show-select"
                tabIndex="0"
                ref={select => {
                  this.select = select;
                }}
              >
                <span className="mdc-select__selected-text">
                  {props.itemsToShow}
                </span>
                <div className="mdc-simple-menu mdc-select__menu">
                  <ul className="mdc-list mdc-simple-menu__items">
                    {itemsToShowValues.map(item => (
                      <li className="mdc-list-item" role="option" tabIndex="0">
                        {item}
                      </li>
                    ))}
                  </ul>
                </div>
              </div>
            </Toolbar.Section>
            <Toolbar.Section align-end className="App-toolbar-section">
              {props.syncState.syncing
                ? <Spinner id="syncing-spinner" size={24} />
                : <IconButton title="Sync items" onClick={props.attemptSync}>
                    <Sync />
                  </IconButton>}
            </Toolbar.Section>
          </Toolbar.Row>
        </Toolbar>
        <div className="flex App-content">
          {props.matches
            ? <Drawer.PermanentDrawer>
                <Toolbar className="mdc-theme--text-primary-on-accent mdc-theme--accent-bg">
                  <Toolbar.Row />
                  <Toolbar.Row>
                    {props.profile.name}
                  </Toolbar.Row>
                </Toolbar>
                <NavItems activeClassName="mdc-permanent-drawer--selected" />
              </Drawer.PermanentDrawer>
            : <Drawer.TemporaryDrawer
                ref={drawer => {
                  this.drawer = drawer;
                }}
              >
                <Drawer.TemporaryDrawerHeader className="mdc-theme--accent-bg">
                  {props.profile.name}
                </Drawer.TemporaryDrawerHeader>
                <Drawer.TemporaryDrawerContent>
                  <NavItems
                    onListClick={() => {
                      this.drawer.MDComponent.open = false;
                    }}
                    activeClassName="mdc-temporary-drawer--selected"
                  />
                </Drawer.TemporaryDrawerContent>
              </Drawer.TemporaryDrawer>}
          <main className="flex-child flex mdc-toolbar-fixed-adjust">
            <AppRouter onChange={this.onRouteChange} />
          </main>
        </div>
        <Dialogs />
      </div>
    );
  }
}

function mapStateToProps({ toasts, itemsToShow, syncState, profile }) {
  return { toasts, itemsToShow, syncState, profile };
}

export default MatchMedia('(min-width: 600px)')(
  connect(mapStateToProps, mapDispatchToProps)(App)
);
