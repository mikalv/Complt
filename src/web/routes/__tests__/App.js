import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

const dialogs = {
  updateItem: {
    visible: false,
  },
  moveItem: {
    visible: false,
  },
};

const syncState = {
  syncing: false,
};

describe('App component', () => {
  it('renders without crashing', () => {
    shallow(<App location={{ pathname: '/inbox' }} toasts={[]} dialogs={dialogs} syncState={syncState} profile={{}}><div /></App>);
  });
  it('renders correctly when the pathname matches a route', () => {
    const component = shallow(<App location={{ pathname: '/inbox' }} toasts={[]} dialogs={dialogs} syncState={syncState} profile={{}}><div /></App>);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly when the pathname does not match a route', () => {
    const component = shallow(<App location={{ pathname: '/some/path/that/does/not/exist' }} toasts={[]} dialogs={dialogs} syncState={syncState} profile={{}}><div /></App>);
    expect(component).toMatchSnapshot();
  });
});
