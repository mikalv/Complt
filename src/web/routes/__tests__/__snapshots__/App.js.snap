exports[`App component renders correctly when the pathname does not match a route 1`] = `
<NavigationDrawer
  autoclose={true}
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="clipped"
  desktopMinWidth={1025}
  drawerHeader={
    <Toolbar
      colored={true}
      component="header"
      prominentTitle={true}
      singleColor={true}
      style={
        Object {
          "backgroundColor": "#ff4081",
        }
      } />
  }
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      Object {
        "component": "a",
        "key": "all",
        "leftIcon": <MdApps
          size={24} />,
        "primaryText": "All",
        "to": "/all",
      },
      Object {
        "component": "a",
        "key": "inbox",
        "leftIcon": <MdInbox
          size={24} />,
        "primaryText": "Inbox",
        "to": "/inbox",
      },
      Object {
        "component": "a",
        "key": "projects",
        "leftIcon": <MdAssignment
          size={24} />,
        "primaryText": "Projects",
        "to": "/projects",
      },
      Object {
        "component": "a",
        "key": "tags",
        "leftIcon": <MdLabel
          size={24} />,
        "primaryText": "Tags",
        "to": "/tags",
      },
      Object {
        "component": "a",
        "key": "today",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Today",
        "to": "/today",
      },
      Object {
        "component": "a",
        "key": "tomorrow",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Tomorrow",
        "to": "/tomorrow",
      },
      Object {
        "component": "a",
        "key": "overdue",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Overdue",
        "to": "/overdue",
      },
      Object {
        "component": "a",
        "key": "week",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Next 7 Days",
        "to": "/week",
      },
      Object {
        "component": "a",
        "key": "account",
        "leftIcon": <MdPerson
          size={24} />,
        "primaryText": "Account",
        "to": "/account",
      },
    ]
  }
  persistentIconChildren="arrow_back"
  position="left"
  tabletDrawerType="clipped"
  tabletMinWidth={768}
  temporaryIconChildren={<MdMenu />}
  toolbarActions={
    Array [
      <ButtonTooltipedInked
        icon={true}
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}>
        <MdSync />
    </ButtonTooltipedInked>,
    ]
  }
  toolbarThemeType="colored"
  toolbarTitleMenu={
    <SelectField
      defaultValue=""
      iconChildren={
        <MdArrowDropDown
          size={24} />
      }
      id="items-to-show-selection"
      itemLabel="label"
      itemValue="value"
      keyboardMatchingTimeout={1000}
      lineDirection="left"
      menuItems={
        Array [
          "All",
          "Not Completed",
          "Completed",
        ]
      }
      position="tl"
      stretchList={true} />
  }
  transitionEnterTimeout={300}
  transitionName="md-cross-fade">
  <Component />
  <SnackbarContainer
    autohide={true}
    autohideTimeout={3000}
    toasts={Array []}
    transitionEnterTimeout={300}
    transitionLeaveTimeout={300}
    transitionName="md-snackbar" />
  <Connect(Component) />
  <Connect(Component) />
</NavigationDrawer>
`;

exports[`App component renders correctly when the pathname matches a route 1`] = `
<NavigationDrawer
  autoclose={true}
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="clipped"
  desktopMinWidth={1025}
  drawerHeader={
    <Toolbar
      colored={true}
      component="header"
      prominentTitle={true}
      singleColor={true}
      style={
        Object {
          "backgroundColor": "#ff4081",
        }
      } />
  }
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      Object {
        "component": "a",
        "key": "all",
        "leftIcon": <MdApps
          size={24} />,
        "primaryText": "All",
        "to": "/all",
      },
      Object {
        "active": true,
        "component": "a",
        "key": "inbox",
        "leftIcon": <MdInbox
          size={24} />,
        "primaryText": "Inbox",
        "to": "/inbox",
      },
      Object {
        "component": "a",
        "key": "projects",
        "leftIcon": <MdAssignment
          size={24} />,
        "primaryText": "Projects",
        "to": "/projects",
      },
      Object {
        "component": "a",
        "key": "tags",
        "leftIcon": <MdLabel
          size={24} />,
        "primaryText": "Tags",
        "to": "/tags",
      },
      Object {
        "component": "a",
        "key": "today",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Today",
        "to": "/today",
      },
      Object {
        "component": "a",
        "key": "tomorrow",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Tomorrow",
        "to": "/tomorrow",
      },
      Object {
        "component": "a",
        "key": "overdue",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Overdue",
        "to": "/overdue",
      },
      Object {
        "component": "a",
        "key": "week",
        "leftIcon": <MdToday
          size={24} />,
        "primaryText": "Next 7 Days",
        "to": "/week",
      },
      Object {
        "component": "a",
        "key": "account",
        "leftIcon": <MdPerson
          size={24} />,
        "primaryText": "Account",
        "to": "/account",
      },
    ]
  }
  persistentIconChildren="arrow_back"
  position="left"
  tabletDrawerType="clipped"
  tabletMinWidth={768}
  temporaryIconChildren={<MdMenu />}
  toolbarActions={
    Array [
      <ButtonTooltipedInked
        icon={true}
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}>
        <MdSync />
    </ButtonTooltipedInked>,
    ]
  }
  toolbarThemeType="colored"
  toolbarTitleMenu={
    <SelectField
      defaultValue=""
      iconChildren={
        <MdArrowDropDown
          size={24} />
      }
      id="items-to-show-selection"
      itemLabel="label"
      itemValue="value"
      keyboardMatchingTimeout={1000}
      lineDirection="left"
      menuItems={
        Array [
          "All",
          "Not Completed",
          "Completed",
        ]
      }
      position="tl"
      stretchList={true} />
  }
  transitionEnterTimeout={300}
  transitionName="md-cross-fade">
  <Component />
  <SnackbarContainer
    autohide={true}
    autohideTimeout={3000}
    toasts={Array []}
    transitionEnterTimeout={300}
    transitionLeaveTimeout={300}
    transitionName="md-snackbar" />
  <Connect(Component) />
  <Connect(Component) />
</NavigationDrawer>
`;
